packages/cosmic-swingset/src/agcc-shim.js

```diff
@@ -0,0 +1,21 @@
+/* global process */
+const { Fail } = assert;
+
+let agccP;
+const { AGVM_FROM_AGD, AGVM_TO_AGD } = process.env;
+if (AGVM_FROM_AGD && AGVM_TO_AGD) {
+  const fd = [parseInt(AGVM_FROM_AGD, 10), parseInt(AGVM_TO_AGD, 10)];
+  for (const f of fd) {
+    Number.isSafeInteger(f) || Fail(`fd ${f} must be a safe integer`);
+    f >= 0 || Fail(`fd ${f} must be non-negative`);
+  }
+  const spec = JSON.stringify({
+    agd: { fd },
+  });
+  agccP = import('./agcc-agvm.js').then(ns => ns.makeAgccFromSpec(spec));
+} else {
+  agccP = import('@agoric/cosmos').then(ns => ns.default);
+}
+
+const dflt = agccP;
+export default dflt;
```

Comment:
crazy suggestion

```suggestion
export const then = Promise.prototype.then.bind(agccP);
```

then in `entrypoint.js`, simply do
```js
import('./agcc-shim.js').then(agcc =>
```